core/src/main/java/org/apache/brooklyn/core/entity/AbstractEntity.java

```diff
@@ -1698,6 +1698,7 @@ public void add(Policy policy) {
             CatalogUtils.setCatalogItemIdOnAddition(AbstractEntity.this, policy);
             policiesInternal.add((AbstractPolicy)policy);
             ((AbstractPolicy)policy).setEntity(AbstractEntity.this);
+            ConfigConstraints.assertValid(policy);
```

Comment:
No, this asserts that the `ConfigConstraints` on the policy are valid. It had to be moved to cope with the situation where some config keys had values that were resolvable only using the *entities* context. So it must happen *after* `setEntity()` is called on the policy. I fixed enrichers too, just for luck.